from snakemake.utils import min_version

##### set minimum snakemake version #####
min_version("8.8.0")

##### Load configuration file #####
configfile: "config/config.yaml"

##### setup report #####
report: "report/workflow.rst"


# Use values from the config file for input and output directories
input_dir = config['data']['raw_reads']
gtf = config['ref']['annotation']

# get sample names based on the .fastq files in the input directory
# samples, = glob_wildcards(f"{input_dir}/{{sample}}.fastq")
# if not samples:
#     raise ValueError(f"No FASTQ files found in {input_dir}")


##### load rules #####


rule all:
    input:
        expand(f"{config['out']['out_dir']}/index_genome.{{i}}.bt2", i=config["index_files"])
        # expand(f"{output_dir}/{{sample}}_aligned.bam", sample=samples)

# include: "rules/all.smk"
# include: "rules/fastq2fasta.smk"
include: "rules/indexing.smk"
# include: "rules/align.smk"

